src/Xamarin.Android.Tools.AndroidSdk/AndroidSdkInfo.cs

```diff
@@ -171,5 +171,20 @@ public static void SetPreferredJavaSdkPath (string path, Action<TraceLevel, stri
 			var sdk = CreateSdk (logger);
 			sdk.SetPreferredJavaSdkPath (path);
 		}
+
+		public static void DetectAndSetPreferredJavaSdkPathToLatest (Action<TraceLevel, string> logger = null)
+		{
+			if (OS.IsWindows)
+				throw new NotSupportedException ("Windows is not supported at this time.");
+
+			logger          = logger ?? DefaultConsoleLogger;
+
+			var latestJdk   = JdkInfo.GetMacOSMicrosoftJdks (logger).FirstOrDefault ();
+			if (latestJdk == null)
+				throw new NotSupportedException ("No preferred Java SDK could be found.");
```

Comment:
`JdkInfo.GetMacOSMicrosoftJdks()` returns a list of all our Microsoft OpenJDK installs, not a list of preferred Java SDKs, right?  This exception message might cause some confusion.